yarn-project/pxe/src/storage/note_data_provider/note_data_provider.ts

```diff
@@ -80,17 +103,23 @@ export class NoteDataProvider implements DataProvider {
     return true;
   }
 
+  /**
+   * Adds multiple notes to the data provider under the specified scope.
+   *
+   * Notes are stored using their index from the notes hash tree as the key, which provides
+   * uniqueness and maintains creation order. Each note is indexed by multiple criteria
+   * for efficient retrieval.
+   *
+   * @param notes - Array of NoteDao objects to store
+   * @param scope - The scope (user/account) under which to store the notes (defaults to ZERO)
+   */
   addNotes(notes: NoteDao[], scope: AztecAddress = AztecAddress.ZERO): Promise<void> {
```

Comment:
```suggestion
  addNotes(notes: NoteDao[], scope: AztecAddress): Promise<void> {
```
I checked, because the zero address there seemed weird, and we always call this with the scope defined so the zero address can be dropped.